sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java

```diff
@@ -728,14 +736,30 @@ else if (mUnknownSourcesDialog != null) {
                 }
             }
 
-             /* If not, open browser to update setup. */
-            if (!mBrowserOpenedOrAborted) {
+            /* If not, open native app (if installed) to update setup, unless it already failed. */
+            String testerAppUpdateSetupFailedMessage = PreferencesStorage.getString(PREFERENCE_KEY_TESTER_APP_UPDATE_SETUP_FAILED_MESSAGE_KEY);
+            boolean testerAppUpdateSetupFailed = testerAppUpdateSetupFailedMessage != null && !testerAppUpdateSetupFailedMessage.isEmpty();
+            if (isAppCenterTesterAppInstalled() && !mTesterAppOpenedOrAborted && !testerAppUpdateSetupFailed) {
+                mTesterAppOpenedOrAborted = DistributeUtils.updateSetupUsingTesterApp(mForegroundActivity, mPackageInfo);
+            }
+
+            /* If the native app could not be opened, use the browser to update setup. */
+            if (!mBrowserOpenedOrAborted && (!mTesterAppOpenedOrAborted || testerAppUpdateSetupFailed)) {
```

Comment:
do you really need 2 different booleans?

Reply:
do you mean what's the reason to have both `mTesterAppOpenedOrAborted` and `testerAppUpdateSetupFailed`? or which two are you referring to because there are 3 here :D

Reply:
those 2 new ones, was wondering if logic could be simplified